packages/tasks_panel/test/misc/e2e-config.spec.ts

```diff
@@ -0,0 +1,472 @@
+        .withExactArgs(testVscode.Uri.file(projectInfo.path))
+        .returns({ uri: { fsPath: "/home/user/test/" } });
+      expect(await collectProjects(wsFolder)).to.be.deep.equal([
+        {
+          wsFolder,
+          project: "my",
+          style: ProjectTypes.CAP,
+        },
+      ]);
+    });
+
+    it("collectProjects - project found, project type is 'com.sap.hana' type", async () => {
+      const wsFolder = "/home/user/test/project/my/";
+      const projectInfo = {
+        path: testVscode.Uri.joinPath(testVscode.Uri.file(wsFolder)).path,
+        type: "com.sap.hana",
+      };
+      sandbox.stub(infoProject, "getProjectInfo").resolves(projectInfo);
+      sandbox.stub(btaExtension.exports.workspaceAPI, "getProjects").resolves([infoProject]);
+      mockExtensions.expects("getExtension").withExactArgs("SAPOSS.app-studio-toolkit").returns(btaExtension);
+      mockWorkspace
+        .expects("getWorkspaceFolder")
+        .withExactArgs(testVscode.Uri.file(projectInfo.path))
+        .returns({ uri: { fsPath: wsFolder } });
+      mockWorkspace.expects("asRelativePath").withExactArgs(wsFolder, false).returns(wsFolder);
+      expect(await collectProjects(wsFolder)).to.be.deep.equal([
+        {
+          wsFolder,
+          project: "",
+          style: ProjectTypes.HANA,
+        },
+      ]);
+    });
+  });
+
+  describe("addTaskDefinition scope", () => {
+    const wsFolder = "/home/user/test/project/my/";
+
+    let mockUpdateConfiguration: SinonMock;
+    beforeEach(() => {
+      mockUpdateConfiguration = sandbox.mock(utils);
+      mockUpdateConfiguration.expects("updateTasksConfiguration").withArgs(wsFolder).resolves();
+    });
+
+    afterEach(() => {
+      mockUpdateConfiguration.verify();
+    });
+
+    const tasks = [
+      {
+        label: "task1",
+        type: "type1",
+        __wsFolder: wsFolder,
+      },
+      {
+        label: "task2",
+        type: "type2",
+        __wsFolder: wsFolder,
+      },
+    ];
+    it("addTaskDefinition - get `tasks` returns `undefined`", async () => {
+      mockWorkspace
+        .expects("getConfiguration")
+        .withExactArgs("tasks", testVscode.Uri.file(wsFolder))
+        .returns(undefined);
+      await addTaskDefinition(wsFolder, tasks);
+      const _tasks = (<any>mockUpdateConfiguration).expectations["updateTasksConfiguration"][0].args[0][1];
+      expect(_tasks).be.deep.equal(tasks);
+    });
+
+    it("addTaskDefinition - update the existing tasks list", async () => {
+      const exisTasks = [{ label: "task3", type: "other" }];
+      mockWorkspace
+        .expects("getConfiguration")
+        .withExactArgs("tasks", testVscode.Uri.file(wsFolder))
+        .returns({ get: () => exisTasks });
+      await addTaskDefinition(wsFolder, tasks);
+      const _tasks = (<any>mockUpdateConfiguration).expectations["updateTasksConfiguration"][0].args[0][1];
+      expect(_tasks).be.deep.equal(concat(exisTasks, tasks));
```

Comment:
in modern javascript you can replace:
`concat(exisTasks, tasks)`

with
`[...exisTasks, ...tasks]`